server/game/cards/10-TEL/AkodoMotivator.js

```diff
@@ -0,0 +1,27 @@
+const _ = require('underscore');
+const DrawCard = require('../../drawcard');
+const AbilityDsl = require('../../abilitydsl');
+
+class AkodoMotivator extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            title: 'Opponent discards an equal number of cards at random',
+            when: {
+                onCardsDiscardedFromHand: (event, context) => {
+                    const discardedFromOwnHand = (event.player === context.player);
+                    const discardedByOpponentsEffect = (event.player === event.context.player.opponent);
+                    const discardedByRingEffect = (event.context.source.type === 'ring');
+                    const discardedByCardEffect = event.context.ability.isCardAbility();
+                    return discardedFromOwnHand && discardedByOpponentsEffect && (discardedByRingEffect || discardedByCardEffect);
+                }
+            },
+            gameAction: AbilityDsl.actions.discardAtRandom(context => ({
+                amount: _.isArray(context.event.cards) ? context.event.cards.length : context.event.amount
```

Comment:
This should be just `context.event.amount`, but I suspect there is a reason why it isn't.... I think we need to fix the reason though, not hack around it :)

Reply:
Yeah the reason is that `amount` doesn't exist when the discard is chosen, and `cards` doesn't exist when the discard is random. I agree that it would be good to make it so that amount (and also cards? could be more complex depending on the current implementation) is always available. I could take a look at that, it doesn't sound like it would be a very difficult change for a newcomer.

Reply:
Yeah, amount should be on the event when the discard is chosen, that's the problem.  If you can find your way round game actions and add that, that would be great.

Reply:
Fixed.